src/generic/FreeAssAlgebra.jl

```diff
@@ -0,0 +1,549 @@
+###############################################################################
+#
+#   FreeAssAlgebra.jl : free associative algebra R<x1,...,xn>
+#
+###############################################################################
+
+export set_exponent_word
+
+###############################################################################
+#
+#   Data type and parent object methods
+#
+###############################################################################
+
+function parent_type(::Type{FreeAssAlgElem{T}}) where T <: RingElement
+   return FreeAssAlgebra{T}
+end
+
+function elem_type(::Type{FreeAssAlgebra{T}}) where T <: RingElement
+   return FreeAssAlgElem{T}
+end
+
+function parent(a::FreeAssAlgElem)
+   return a.parent
+end
+
+function base_ring(a::FreeAssAlgebra)
+   return a.base_ring
+end
+
+function symbols(a::FreeAssAlgebra)
+   return a.S
+end
+
+function nvars(a::FreeAssAlgebra)
+   return length(a.S)
+end
+
+function length(a::FreeAssAlgElem)
+   return a.length
+end
+
+function check_parent(a::FreeAssAlgElem{T}, b::FreeAssAlgElem{T}, throw::Bool = true) where T <: RingElement
+   b = parent(a) != parent(b)
+   b & throw && error("Incompatible polynomial rings in polynomial operation")
```

Comment:
Probably shouldn't refer to it as a polynomial ring.

Reply:
sure.